### PR TITLE
Implementing SharedDirectory,applyStashedOp()

### PR DESCRIPTION
**Pull request by dannimad**
https://dev.azure.com/fluidframework/internal/_workitems/edit/2174

## Description

> This PR will contain all sharedDirectory applyStashedOp operations
> For now, it us just a draft PR so I can receive early feedback on the overall design

## TODO

- [x] Overall structure
- [x] Set handler
- [x] Delete handler
- [x] Clear handler
- [x] Create subdirectory handler
- [x] Delete subdirectory handler
- [x]  Fix doc build error
- [x] Unit tests

### CODE REVIEW

File: packages/dds/map/src/directory.ts [hunk 1]
```diff
@@ -1744,7 +1873,7 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
     /**
      * Clear all keys in memory in response to a remote clear, but retain keys we have modified but not yet been ack'd.
      */
-    private clearExceptPendingKeys() {
+    private clearExceptPendingKeys(local: boolean) {
```
Comment (scarlettjlee) [inline packages/dds/map/src/directory.ts:1876]: What is "local" used for?

Comment (dannimad) [reply packages/dds/map/src/directory.ts:1876]: It is used in clearCore to send that information in the emit but I didn't send it, (is set to false rn) which is a bug. I will change it thank you

**Review by wes-carlson — COMMENTED** [review]
Looks good, but it looks like you're only testing set(). I think you should add some more coverage, including subdirectory creation/deletion.

**Review by wes-carlson — APPROVED** [review]
:shipit:

**Pull request merged**